mifosng-android/src/main/java/com/mifos/api/datamanager/DataManagerSearch.java

```diff
@@ -17,14 +25,53 @@
 public class DataManagerSearch {
 
     public final BaseApiManager baseApiManager;
+    public final DatabaseHelperClient mDatabaseHelperClient;
+    public final DatabaseHelperGroups mDatabaseHelperGroup;
+    public final DatabaseHelperCenter mDatabaseHelperCenter;
+    public final DatabaseHelperLoan mDatabaseHelperLoan;
+    public final DatabaseHelperSavings mDatabaseHelperSaving;
 
     @Inject
-    public DataManagerSearch(BaseApiManager baseApiManager) {
+    public DataManagerSearch(BaseApiManager baseApiManager,
+                             DatabaseHelperClient databaseHelperClient,
+                             DatabaseHelperGroups databaseHelperGroup,
+                             DatabaseHelperCenter databaseHelperCenter,
+                             DatabaseHelperLoan databaseHelperLoan,
+                             DatabaseHelperSavings databaseHelperSaving) {
         this.baseApiManager = baseApiManager;
+        mDatabaseHelperClient = databaseHelperClient;
+        mDatabaseHelperGroup = databaseHelperGroup;
+        mDatabaseHelperCenter = databaseHelperCenter;
+        mDatabaseHelperLoan = databaseHelperLoan;
+        mDatabaseHelperSaving = databaseHelperSaving;
     }
 
-    public Observable<List<SearchedEntity>> searchResources(String query, String resources,
-                                                            Boolean exactMatch) {
-        return baseApiManager.getSearchApi().searchResources(query, resources, exactMatch);
+    public Observable<List<SearchedEntity>> searchResources(final String query, String resources,
+                                                            final Boolean exactMatch) {
+        switch (PrefManager.getUserStatus()) {
+            case 0:
+                return baseApiManager.getSearchApi().searchResources(query, resources, exactMatch);
+            case 1:
+                if (resources.equals(Constants.ENTITY_TYPE_CLIENTS)) {
+                    return mDatabaseHelperClient.readAllClientsByQuery(query, exactMatch);
+                } else if (resources.equals(Constants.ENTITY_TYPE_GROUPS)) {
+                    return Observable.merge(mDatabaseHelperGroup.readAllGroupsByQuery(query, exactMatch),
```

Comment:
@therajanmaurya I am waiting for your reply. I have tried by myself but still, I was not able to resolve the error. 
The error is same - 
https://github.com/openMF/android-client/pull/741#discussion_r131520069